angular2-jwt.ts

```diff
@@ -119,29 +131,14 @@ export class AuthHttp {
 
     // from this point url is always an instance of Request;
     let req: Request = url as Request;
-    let token: string & Promise<string> = this.config.tokenGetter();
-    if(token&&token.then) {
-      return Observable.fromPromise(token)
-          .flatMap((jwtToken: string) => this.requestWithToken(req, jwtToken));
+    let token: string | Promise<string> = this.config.tokenGetter();
+    if (token instanceof Promise) {
+      return Observable.fromPromise(token).flatMap((jwtToken: string) => this.requestWithToken(req, jwtToken));
```

Comment:
This should be mergeMap, not flatMap. flatMap is an alias for mergeMap, and I'm not confident it'll be around forever.